### PR TITLE
Forward support for Float module

### PR DESCRIPTION
**Pull request by dbuenzli**
/cc @nojb @chris00 

I reimplemented the float array stuff in terms of `Array`. Not sure if that's a good idea. Also I suspect the creation primitive `caml_floatarray_create` does not initializes the array with `0.0` while this implementation does.

### CODE REVIEW

**Comment by ghost** [comment]
That seems fine to me and in general it seems acceptable to me if emulated functions are less efficient.

**Review by ghost — APPROVED** [review]

File: src/float.ml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+let equal x y = compare x y = 0
+external seeded_hash_param : int -> int -> int -> float -> int
+                           = "caml_hash" [@@noalloc]
+let hash x = seeded_hash_param 10 100 0 x
+
+module Array = struct
+  type t = float array
+  let create len = Array.make len 0.0
+  let length = Array.length
+  let get = Array.get
```
Comment (Chris00) [inline src/float.ml]: Wouldn't a type annotation make the compiler select the fast implementation?

Comment (ghost) [reply src/float.ml]: Indeed

**Comment by dbuenzli** [comment]
Thanks ! Added type annotations in `Array`.

**Comment by dbuenzli** [comment]
Let's wait for the outcome of https://github.com/ocaml/ocaml/pull/2005 before merging this.

**Comment by dbuenzli** [comment]
Rebased and merged.

**Pull request merged**